zebra-network/src/peer_set/set.rs

```diff
@@ -497,6 +508,15 @@ where
                     // - check for any errors that happened right after the handshake
                     trace!(?key, "got Change::Insert from Discover");
                     self.remove(&key);
+
+                    // drop the new peer if there are already `MAX_CONNS_PER_IP` peers with
+                    // the same IP address in the peer set
+                    // this is skipped for tests so we can mock peer connections
+                    #[cfg(not(test))]
+                    if self.num_peers_with_ip(key.ip()) > crate::constants::MAX_CONNS_PER_IP {
+                        continue;
```

Comment:
Nitpick: I prefer explicit drops, particularly in security fixes.

```suggestion
                        std::mem::drop(svc);
                        continue;
```